### PR TITLE
Json serialization fixes

### PR DESCRIPTION
**Pull request by potyl**
This branch fixes serialization issues that cause stetho to die when we try to send back NaN, Infinity or -Infinity. These values are not part of the JSON specification. What this patch does is to serialize them as String since JavaScript will accept them back as Doubles.

### CODE REVIEW

File: stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java [hunk 1]
```diff
@@ -273,6 +273,18 @@ private Object getJsonValue(Object value, Class<?> clazz, Field field)
       return convertValue(value, JSONObject.class);
     }
     // hmm we should be able to directly serialize here...
```
Comment (jasta) [inline stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java]: this command should probably be moved to after this new `if` statement

File: stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java [hunk 2]
```diff
@@ -101,7 +101,7 @@
       Object value = jsonObject.opt(field.getName());
       Object setValue = getValueForField(field, value);
       try {
-        field.set(instance, getValueForField(field, value));
+        field.set(instance, setValue);
```
Comment (jasta) [inline stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java]: good catch :)

**Comment by jasta** [comment]
only a minor comment, looks good otherwise

File: stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java (hunk 1 above)
Comment (potyl) [inline stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java]: I don't understand this comment. Do you mean "this comment" ???

Comment (jasta) [inline stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java]: yes sorry, I meant this comment.  it should be just before the last line's return statement.

Comment (potyl) [inline stetho/src/main/java/com/facebook/stetho/json/ObjectMapper.java]: Yes. I'll change it.

**Pull request merged**